.github/workflows/lint.yml

```diff
@@ -10,32 +10,36 @@ name: Golang Linter
 
 on:
   pull_request:
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
   push:
     branches:
       - main
       - v[0-9]+.[0-9]+.x-celestia
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
+
 jobs:
   golangci:
     name: golangci-lint
     runs-on: ubuntu-latest
     timeout-minutes: 8
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v4
-        with:
-          go-version: '1.22'
-      - uses: technote-space/get-diff-action@v6
+      - uses: actions/checkout@v4
+      
+      - uses: actions/setup-go@v5
         with:
-          PATTERNS: |
-            **/**.go
-            go.mod
-            go.sum
-      - uses: golangci/golangci-lint-action@v3.7.0
+          go-version-file: 'go.mod'
+
+      - uses: golangci/golangci-lint-action@v4.0.0
         with:
           # Required: the version of golangci-lint is required and
           # must be specified without patch version: we always use the
           # latest patch version.
           version: v1.55.2
```

Comment:
[optional] we could get rid of this comment b/c it's incorrect and bump to latest golangci-lint https://github.com/golangci/golangci-lint/releases/tag/v1.56.2

but we can def do this in a follow up